### PR TITLE
(#18) Assertion

### PR DESCRIPTION
**Pull request by llorllale**
This PR is for #18:

* Added `Assertion`
* `Throws` now declares a generic parameter in order to be compatible
  with `Assertion`
* Left puzzle to refactor existing tests with `Assertion`
* Left puzzle to continue refining `Assertion`
* Took the liberty of gitignoring `nbproject` because I use Netbeans

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#49](https://github.com/llorllale/cactoos-matchers/issues/49) is now in scope, role is `REV`

File: src/main/java/org/llorllale/cactoos/matchers/Assertion.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    private final Matcher<T> matcher;
+
+    /**
+     * Ctor.
+     * @param reason Reason for refuting this assertion
+     * @param test The behaviour to test
+     * @param matcher Matcher to test behaviour
+     */
+    public Assertion(
+        final String reason, final Scalar<T> test, final Matcher<T> matcher
```
Comment (dgroup) [inline src/main/java/org/llorllale/cactoos/matchers/Assertion.java]: @llorllale, can we use the following formatting for the methods and ctors always?

Comment (llorllale) [reply src/main/java/org/llorllale/cactoos/matchers/Assertion.java]: @dgroup totally agree

Comment (llorllale) [reply src/main/java/org/llorllale/cactoos/matchers/Assertion.java]: @dgroup well, not *always* - only when the line length limit is exceeded (obviously)

**Comment by llorllale** [comment]
@dgroup you want to review this?

File: src/main/java/org/llorllale/cactoos/matchers/Assertion.java (hunk 1 above)
Comment (dgroup) [reply src/main/java/org/llorllale/cactoos/matchers/Assertion.java]: Yes, for that's purpose only.

**Comment by dgroup** [comment]
@llorllale, yes, looking.

File: src/main/java/org/llorllale/cactoos/matchers/Assertion.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    /**
+     * Description of this assertion's refutation.
+     * @return Refutation
+     */
+    private Description description() {
+        final Description description = new StringDescription();
+        description.appendText(this.reason)
+            .appendText("\nExpected: ")
```
[the remark this answers is not in the document]
Comment (llorllale) [reply src/main/java/org/llorllale/cactoos/matchers/Assertion.java]: @dgroup fixing

File: src/test/java/org/llorllale/cactoos/matchers/AssertionTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ * @since 1.0.0
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class AssertionTest {
+    /**
+     * Assertion can be affirmed if the operation being tested matches.
+     */
+    @Test
+    public void affirmIfResultMatches() {
+        final String test = "abc123";
```
[the remark this answers is not in the document]
Comment (llorllale) [reply src/test/java/org/llorllale/cactoos/matchers/AssertionTest.java]: @dgroup it's a way of declaring that this is the expected result. I'll rename to `expected`

**Comment by llorllale** [comment]
@dgroup done, please check

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#49](https://github.com/llorllale/cactoos-matchers/issues/49) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @dgroup[/z](https://www.0crat.com/u/dgroup)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Pull request merged**